sponsorship/src/test/java/com/kyriba/conference/sponsorship/dao/SponsorRepositoryTest.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************************
+ * Copyright 2000 - 2019 Kyriba Corp. All Rights Reserved.                  *
+ * The content of this file is copyrighted by Kyriba Corporation            *
+ * and can not be reproduced, distributed, altered or used in any form,     *
+ * in whole or in part.                                                     *
+ *                                                                          *
+ * Date          Author         Changes                                     *
+ * 2019-07-12    M-ASL          Created                                     *
+ *                                                                          *
+ ****************************************************************************/
+package com.kyriba.conference.sponsorship.dao;
+
+import com.github.database.rider.core.api.connection.ConnectionHolder;
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.core.api.dataset.ExpectedDataSet;
+import com.github.database.rider.junit5.api.DBRider;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.ActiveProfiles;
+
+
+/**
+ * @author Aliaksandr Samal
+ */
+@DataJpaTest
+@DBRider
+@ActiveProfiles("test")
+class SponsorRepositoryTest
+{
+  @Autowired
+  private SponsorRepository sponsorRepository;
+  @Autowired
+  private TestEntityManager entityManager;
+
+  @Autowired
+  protected JdbcTemplate jdbcTemplate;
+
+  private ConnectionHolder connectionHolder = () -> jdbcTemplate.getDataSource().getConnection();
+
+
+  //todo this test is failing for unkown reason
+  @Test
+  @DataSet("datasets/empty.xml")
+  @ExpectedDataSet("datasets/sponsor-created.xml")

```

Comment:
@xpinjection , please, could you look why `@ExpectedDataSet` doesn't work here?

Reply:
Despite of documentation for JUnit 5, extensions ordering doesn't work as expected. To make it work you could commit transaction as the last statement in test with entityManager.getEntityManager().getTransaction().commit();

Reply:
Thanks, it works